identity/webapp/pages/api/registration-form.ts

```diff
@@ -0,0 +1,29 @@
+import type { NextApiRequest, NextApiResponse } from 'next';
+import { generateNewToken, decodeToken } from '../../src/utility/jwt-codec';
+
+export default (req: NextApiRequest, res: NextApiResponse): void => {
+  if (req.method === 'POST') {
+    const {
+      state,
+      redirectUri,
+      firstName,
+      lastName,
+      termsAndConditions,
+      sessionToken,
+    } = req.body;
+
+    const decodedToken = decodeToken(sessionToken);
+    const formData = { firstName, lastName, termsAndConditions };
+
+    if (typeof decodedToken !== 'string') {
+      const newToken = generateNewToken(decodedToken, state, formData);
+
+      res.redirect(
+        308,
```

Comment:
Why a 308 here? I think browsers will cache this very aggressively, which might cause issues

Reply:
Yeah, not sure why I said 301 lol 🙃  This is interesting - it's definitely not a permanent redirect, so looking at [this](https://developer.mozilla.org/en-US/docs/Web/HTTP/Redirections#temporary_redirections) it's not actually clear that there's anything which is a good fit. This is, I think, a good clue that we should tweak the approach...

- We're making a POST to our endpoint with our form data (this is fine and good)
- We want to Do Some Stuff with that data and then POST to the `/continue` endpoint of Auth0 (I think?) as per [these docs](https://auth0.com/docs/customize/actions/flows-and-triggers/login-flow/redirect-with-actions#pass-information-back-to-the-action)
- We are currently trying to make that POST by giving the client a redirect to it, but that seems hard. Looking into it further, it's explicitly disallowed by [the spec](https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html#sec10.3)!
  
  > in response to a request other than GET or HEAD, the user agent MUST NOT automatically redirect the request unless it can be confirmed by the user
- Solution: I think we should just make that request ourselves in the server-side code (ie with axios or whatever), and then redirect with a 302 to the account page (or an error page).

Reply:
I think I get what you mean, so line 21 is currently a redirect - we need that to be a post so we can use something like `axios.post()` then redirect to error/success page